features/vaadin-opennms-pluginmanager/src/main/java/org/opennms/features/pluginmgr/vaadin/config/opennms/PluginManagerAdminApplication.java

```diff
@@ -0,0 +1,214 @@
+@Theme("opennms")
+@Title("Plugin Manager Administration")
+@SuppressWarnings("serial")
+public class PluginManagerAdminApplication extends UI {
+	private static final Logger LOG = LoggerFactory.getLogger(PluginManagerAdminApplication.class);
+
+	/*- VaadinEditorProperties={"grid":"RegularGrid,20","showGrid":true,"snapToGrid":true,"snapToObject":true,"movingGuides":false,"snappingDistance":10} */
+
+	//headerLinks map of key= name and value=url for links to be placed in header of page
+    private Map<String,String> headerLinks;
+    
+	private OnmsHeaderProvider m_headerProvider;
+	private String m_headerHtml;
+	private VaadinRequest m_request;
+	private VerticalLayout m_rootLayout;
+
+	private SessionPluginManager sessionPluginManager;
+	
+	public SessionPluginManager getSessionPluginManager() {
+		return sessionPluginManager;
+	}
+
+	public void setSessionPluginManager(SessionPluginManager sessionPluginManager) {
+		this.sessionPluginManager = sessionPluginManager;
+	}
+
+	/**
+	 * headerLinks map of key= name and value=url for links to be placed in header of page
+	 * @return
+	 */
+	public Map<String,String> getHeaderLinks() {
+		return headerLinks;
+	}
+
+	/**
+	 * @param headerLinks map of key= name and value=url for links to be placed in header of page
+	 */
+	public void setHeaderLinks(Map<String,String> headerLinks) {
+		this.headerLinks = headerLinks;
+	}
+
+	// imported ideas from org.opennms.features.vaadin.nodemaps.internal.NodeMapsApplication
+	public OnmsHeaderProvider getHeaderProvider() {
+		return m_headerProvider;
+	}
+
+	public void setHeaderProvider(OnmsHeaderProvider headerProvider) {
+		this.m_headerProvider = headerProvider;
+	}
+
+	public void setHeaderHtml(final String headerHtml) {
+		m_headerHtml = headerHtml;
+	}
+
+	private void addHeader() {
+		if (m_headerProvider != null) {
+			try {
+				URL pageUrl = Page.getCurrent().getLocation().toURL();
+				setHeaderHtml(m_headerProvider.getHeaderHtml(new HttpServletRequestVaadinImpl(m_request, pageUrl)));
+			} catch (final Exception e) {
+				LOG.error("failed to get header HTML for request " + m_request.getPathInfo(), e.getCause());
+			}
+		}
+		if (m_headerHtml != null) {
+			InputStream is = null;
+			try {
+				is = new ByteArrayInputStream(m_headerHtml.getBytes());
+				final CustomLayout headerLayout = new CustomLayout(is);
+				headerLayout.setWidth("100%");
+				headerLayout.addStyleName("onmsheader");
+				m_rootLayout.addComponent(headerLayout);
+			} catch (final IOException e) {
+				closeQuietly(is);
+				LOG.debug("failed to get header layout data", e);
+			}
+		}
+	}
+
+	private void closeQuietly(InputStream is) {
+		if (is != null) {
+			try {
+				is.close();
+			} catch (final IOException closeE) {
+				LOG.debug("failed to close HTML input stream", closeE);
+			}
+		}
+	}
+
+
+
+	/* (non-Javadoc)
+	 * @see com.vaadin.ui.UI#init(com.vaadin.server.VaadinRequest)
+	 */
+	@Override
+	public void init(VaadinRequest request) {
+
+		m_request = request;
```

Comment:
Please do not store the request initially creating the Vaadin Application as a member variable.